source/dub/packagesupplier.d

```diff
@@ -185,13 +185,30 @@ class RegistryPackageSupplier : PackageSupplier {
 	void fetchPackage(Path path, string packageId, Dependency dep, bool pre_release)
 	{
 		import std.array : replace;
+		import std.net.curl : CurlException;
 		Json best = getBestPackage(packageId, dep, pre_release);
 		if (best.type == Json.Type.null_)
 			return;
 		auto vers = best["version"].get!string;
 		auto url = m_registryUrl ~ Path(PackagesPath~"/"~packageId~"/"~vers~".zip");
 		logDiagnostic("Downloading from '%s'", url);
-		download(url, path);
+		foreach(i; 0..3) {
+			try{
+				download(url, path);
+				return;
+			}
+			catch(HTTPStatusException e) {
+				if (e.status == 404) {
+					logDebug("Failed to download package %s from %s (404)", packageId, url);
+					return;
```

Comment:
I'm not sure how the outer code will behave if nothing downloaded, but to be consistent with the previous behavior, this would also have to throw instead of returning.